doc/architecture/architecture.rst

```diff
@@ -1,24 +1,254 @@
+
+#. **Ostro OS is pre-compiled and pre-configured**: Ostro OS leverages the
+   `Yocto Project`_ build system tools and is provided as a pre-compiled and pre-configured
+   distribution specifically tuned for IoT device development. Generating and compiling your own
+   Ostro OS image is very fast since the build system is configured to use the Shared
+   State Cache (SSTATE) mechanism offered by the Ostro OS SSTATE server.
+   In addition, pre-compiled reference images are available from the
+   `Ostro Project download server`_ to get you started in no time.
+
+Ostro OS Architecture Stack
+===========================
+
+One way to describe Ostro OS architecture is to build a stack from the
+hardware platform up to an IoT Application layer. Such stack is
+illustrated here.
+
+.. image:: images/ostro_os_architecture.svg
+   :width: 400px
+
+The following sections examine each layer in more details.
+
+IoT Applications
+----------------
+The IoT applications layer consist of all applications that use the
+underlying platform. Ostro OS itself currently does not
+implement any specific IoT use-cases via applications, but
+`application sample recipes`_ are provided to demonstrate the OS frameworks and services.
+
+Programming Interfaces
+-----------------------
+
+Ostro OS provides various application runtimes offering more flexibility and
+options to application developers to implement their IoT solution.
+
+* `Soletta Project`_ is an open-source
+  framework for making IoT applications. The project provides libraries
+  to make it quick and easy to write software for IoT devices. The
+  applications can we written using flow-based programming (FBP) or
+  more traditionally as C applications linking to Soletta's C-based
+  platform APIs.
+
+* `Node.JS`_ is a popular JavaScript runtime to
+  run (IoT) web applications. Ostro OS provides the Node.JS runtime
+  and selected JavaScript APIs (as Node.JS modules) to build IoT
+  applications. The set of APIs includes, for example, a Javascript API for
+  the `Open Connectivity Foundation`_ (OCF) specifications. The JavaScript
+  API set is kept aligned with the API set provided by Soletta.
+
+* Ostro OS also provides support for the popular IoT-space development
+  languages Python and Java\* (OpenJDK8).
+
+Frameworks
+----------
+
+Frameworks provide an abstraction to platform services, making
+application deployment easier. The Ostro Application
+Framework :ref:`application-framework` provides tooling for
+application writers to get the applications running in an Ostro OS
+based image.
+
+The Ostro Application framework implements a `systemd.generator`_ that
+parses the application manifest file to automatically generate a systemd service file.
+
+Ostro OS also pre-integrates `IoTivity`_ framework. IoTivity is an open-source
+project that implements the full OCF specification by `Open Connectivity Foundation`_. In
+addition to `IoTivity`_, another implementation of the specification API is also
```

Comment:
don't go deep into details here, rephrase as "This connectivity framework is also available from Soletta and Node.js".

shouldn't matter Soletta is a different implementation, there is an ongoing talk to make IoTivity use Soletta in the core, nothing decided yet, but it may happen.